shared/utils/logic.ts

```diff
@@ -0,0 +1,336 @@
+import { AllowMyInfoBase, FieldBase, FieldResponse } from '../types'
+
+import {
+  FormCondition,
+  FormDto,
+  PreventSubmitLogic,
+  FormLogic,
+  LogicType,
+  ShowFieldLogic,
+  LogicConditionState,
+} from '../types/form'
+
+export type PickLogicSubset<T extends FormDto = FormDto> = Pick<
+  T,
+  | '_id'
+  | 'form_logics'
+  | 'form_fields'
+  | 'responseMode'
+  | 'form_logics'
+  | 'form_fields'
+>
+
+export interface ClientField extends AllowMyInfoBase, FieldBase {
+  _id?: string
+  isVerifiable?: boolean
+  fieldValue: string
+
+  getQuestion(): string
+}
+
+export type FieldIdSet = Set<string>
```

Comment:
This type does not seem to be used anywhere

Reply:
Actually its used in the helper functions of logic.ts for example `allConditionsExist`

Reply:
We have a duplicate type, can probably just combine them and have the type in this file be the source of truth